src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -3,61 +3,223 @@
+                                                         'but is not a folder '
+            else:
+                assert args.destination_folder.parent.is_dir(), '*ERROR* Cannot create project folder ' \
+                                                                f'{args.destination_folder}, parent is not an ' \
+                                                                'existing folder'
+                args.destination_folder.mkdir()
+            args.destination_folder = args.destination_folder.absolute().resolve()
+        else:
+            args.destination_folder = pathlib.Path.cwd()
+
         if args.uvmhome:
             if args.uvm is not None:
                 self.parser.error('Conflicting options --uvm and --use-uvm-home used')
-            args.uvm = 'ENV-UVM_HOME'
+            args.uvm = pathlib.Path('ENV-UVM_HOME')
         elif args.uvm is not None:
-            if not os.path.isdir(args.uvm):
+            uvm_path = pathlib.Path(args.uvm)
+            if not uvm_path.is_dir():
                 self.parser.error(f'UVM home \'{args.uvm}\' must be a folder')
-            if not os.path.isfile(os.path.join(args.uvm, 'src/uvm_macros.svh')):
+            if not uvm_path.joinpath('src/uvm_macros.svh').is_file():
                 self.parser.error(f'Could not find uvm_macros.svh in \'{args.uvm}/src\'')
-            if not os.path.isfile(os.path.join(args.uvm, 'src/uvm_pkg.sv')):
+            if not uvm_path.joinpath('src/uvm_pkg.sv').is_file():
                 self.parser.error(f'Could not find uvm_pkg.sv in \'{args.uvm}/src\'')
-        return args
+            args.uvm = uvm_path
+
+        if args.uvmlib is None:
+            args.uvmlib = args.worklib
+
+        if args.rel_path_root:
+            args.rel_path_root = [pathlib.Path(folder).absolute().resolve() for folder in args.rel_path_root]
 
-    def parse_args_and_file(self, parse_file):
-        args = self.parse_args()
         ArgsAndFileParser.options = args
-        destination = args.destination_folder if args.destination_folder is not None else os.getcwd()
-        entries = parse_file(args.input_file)
-        return args.project_name, args.input_file, destination, entries
+        return args
+
+    @staticmethod
+    def parse_input_file(parse_file):
+        args = ArgsAndFileParser.options
+        # parse_file = ArgsAndFileParser.get_file_parser()
+        destination = args.destination_folder if args.destination_folder is not None else pathlib.Path.cwd()
+        if parse_file is not None:
+            entries = parse_file(args.input_file)
+            return args.project_name, args.input_file, destination, entries
+        # If the parser is None, we assume that input_file contains a (list of) HDL files
+        entries = {pathlib.Path(entry).absolute().resolve(): args.worklib for entry in args.input_file}
+        return args.project_name, None, destination, entries
```

Comment:
That code is gone now.

Reply:
The 2nd element was the name of the input file which contains a list of design files (see line 146 above). If the design files are listed on the command line, there is no file with the list of files (.f, .csv ...) hence None.

This code was rewritten since.